unic-langid-impl/src/lib.rs

```diff
@@ -39,18 +41,23 @@ impl LanguageIdentifier {
             None
         };
 
-        let mut vars = Vec::with_capacity(variants.len());
-        for variant in variants {
-            vars.push(subtags::parse_variant_subtag(variant.as_ref())?);
-        }
-        vars.sort();
-        vars.dedup();
+        let variants = if !variants.is_empty() {
+            let mut vars = Vec::with_capacity(variants.len());
+            for variant in variants {
```

Comment:
Any reason not to prefer a map/collect?

Reply:
visually the alternative is less aesthetic I think:

```rust
            let mut result = variants.into_iter().map(|v| {
                subtags::parse_variant_subtag(v.as_ref())
            }).collect::<Result<Vec<TinyStr8>, parser::errors::ParserError>>()?;
```

but functionally it is better, so I'll switch.